### ISSUE TITLE
Microsoft.VisualStudio.Component.CoreEditor not found

### ISSUE THREAD

**Issue by xoofx** [dotnet/project-system#4462]
Hey,

I have compiled latest master and I'm trying to install the ProjectSystem.vsixmanifest with a different packageid (+different id for CSharp projects, I removed VB...etc.)

But when trying to install the result vsix from debug folder, I'm getting the following exception from the logs:

````
 Extension cannot be installed to the following products due to missing prerequisites:
 	Visual Studio Enterprise 2019 Preview
 		-------------------------------------------------------
 		Identifier   : Microsoft.VisualStudio.Component.CoreEditor
 		Name         : Visual Studio core editor
 		Version      : [15.0,16.0)
 		Error        : The prerequisite version specified does not match the version installed
```

I can't find anything about this `Microsoft.VisualStudio.Component.CoreEditor` 
How can I install it? Why is is not part of my existing VS2019 Preview?

**Comment by xoofx** [comment]
I modified the prerequisite for `Microsoft.VisualStudio.Component.CoreEditor` to change from `[15.0, 16.0)` to `[15.0, 17.0)` and it seems to go through a this problem

```
    <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,17.0)" DisplayName="Visual Studio core editor" />  
```
But now getting another exception in the install log:

```
14-Jan-19 23:23:33 - Beginning to install extension to Visual Studio Enterprise 2019 Preview...
14-Jan-19 23:23:34 - Install Error : System.InvalidOperationException: A value for 'Component' needs to be specified in the catalog.
   at Microsoft.VisualStudio.ExtensionManager.PackageInstaller.LoadComponent(IDependencyGraph dGraph, Boolean& isProductComponent)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.PerformSetupEngineInstall(InstallableExtensionImpl extension, Boolean installPerMachine, Boolean isPackComponent, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, IProgress`1 progress, InstallFlags installFlags, AsyncOperation asyncOp, Version targetedVsVersion, IInstalledExtension& newExtension)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallInternal(InstallableExtensionImpl extension, InstallFlags installFlags, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, IProgress`1 progress, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.BeginInstall(IInstallableExtension installableExtension, InstallFlags installFlags, AsyncOperation asyncOp, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallWorker(IInstallableExtension extension, InstallFlags installFlags, AsyncOperation asyncOp)
```

This one doesn't give much context, so not sure what's going on here...

**Comment by jmarolf** [comment]
So, this is essentially a VSIX bug. The project system VSIXes are "special" in that they have manifest information that identifies them as "System Components".  If you were to look at the `extension.vsixmanifest` file inside `artifacts\Debug\VSSetup\ProjectSystem.vsix` you would see an entry like this:

```XML
  <Installation SystemComponent="true" Experimental="true">
    <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.Pro" />
    <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VSWinDesktopExpress" />
    <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VWDExpress" />
    <InstallationTarget Version="[15.0,]" Id="Microsoft.VisualStudio.VSWinExpress" />
  </Installation>
```

`SystemComponent="true"` means that this isn't a _real_ vsix but a core VS component that needs to be treated differently by the VS installer.  This entry here makes double-click install not work because the default VSIX installer (_not_ the VS installer) doesn't know how to update core VS components.

`Experimental="true"` means that this VSIX is allowed to install and override existing VS components (the currently installed project system bits).  This is orthogonal to your problem but I thought I would explain for completeness.

I will need to test it, but I believe removing this line will turn `ProjectSystem.vsix` into a normal vsix.
https://github.com/dotnet/project-system/blob/1038ba9e8c0760a9fc7adf89fdc18e339f0c2abd/src/ProjectSystemSetup/ProjectSystemSetup.csproj#L15

**Comment by xoofx** [comment]
Thanks! So I tried to remove this line and also changed the installation path `<ExtensionInstallationFolder>` to a different folder, but I'm getting the same error... 

I have also tried to remove both `SystemComponent="true"` and `Experimental="true"` as I don't really want to override the existing component but would like to add another one.

The context is that I would like to leverage the project system to create a new project system for a toy prototype language and I'm trying to derive it from the CSharp project system (but with a different proj extension...etc.)

So basically, I have modified the `system\src\ProjectSystemSetup\source.extension.vsixmanifest` to change the Id and change the GUID in `CSharpProjectSystemPackage` and in `ManagedProjectSystemPackage` but maybe that's not enough... I can see that there are others GUIID in the project that I could need to change... If you have any guidance on that....

**Comment by clarkcant** [comment]
> I modified the prerequisite for Microsoft.VisualStudio.Component.CoreEditor to change from [15.0, 16.0) to [15.0, 17.0) and it seems to go through this problem
>     <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,17.0)" DisplayName="Visual Studio core editor" />  

I know you've closed the issue, but would you mind sharing how you can modify the prerequisites? Is that in a file somewhere?

**Comment by cilerler** [comment]
@clarkcant look into the file called *extension.vsixmanifest* in the VSIX file.  VSIX file is a simple ZIP file that you can extract the content of it.
